Aleksandrova_Vilena/2/checker/spec/args_parser_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'rspec'
+
+require_relative '../lib/helper'
+
+RSpec.describe ArgsParser do
+  subject { described_class.parse(args) }
+  let(:args) { %w[-n test_out.csv] }
+
+  it ':subdomains' do
+    expect(subject).to eq Hash[:subdomains, false]
+  end
+end
+
+RSpec.describe ArgsParser do
+  subject { described_class.parse(args) }
+  let(:args) { %w[-r test_out.csv] }
+  it ':opensource' do
+    expect(subject).to eq Hash[:opensource, true]
+  end
+end
```

Comment:
Нет необходимости многократно описывать `describe class`. Если нужный разные контексты (аргументы), то делается это так
```suggestion
describe ArgsParser do
  subject { described_class.parse(args) }
  let(:args) { %w[some standard args] }
  it 'should parse some args' do 
    expect(subject).to eq ...
  end
  
  context 'with other arg' do
    let(:args) { %w[some other args] }
    it 'should parse other args' do 
      expect(subject).to eq ...
    end
  end
end
```